### PR TITLE
[MIG] migrate business_requirement_deliverable_project from 10.0 to 12.0

### PR DESCRIPTION
**Pull request by ruter-lyu**

### CODE REVIEW

File: business_requirement_deliverable_project/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    'summary': 'Create projects and tasks directly from'
+            ' the Business Requirement and Resources lines',
+    'version': '12.0.1.0.0',
+    'website': 'https://github.com/OCA/business-requirement',
+    "author": "Elico Corp, Odoo Community Association (OCA)",
+    'depends': [
+        'business_requirement_deliverable',
+        'project',
+        'hr_timesheet',
+        'project_parent'
```
Comment (elicoidal) [inline business_requirement_deliverable_project/__manifest__.py:15]: You probably have to think about migrating this one.

File: business_requirement_deliverable_project/models/project.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            from_project = True
+        default_uom = self.env.user and self.env.user.company_id \
+            and self.env.user.company_id.project_time_mode_id.id
+        if not default_uom:
+            raise ValidationError(
+                _("""Please set working time default unit in project
+                    config settings"""))
+        lines = self.env['business.requirement.resource']
+        for br in br_ids:
+            if br.state not in ['stakeholder_approval', 'cancel', 'done']:
```
Comment (elicoidal) [inline business_requirement_deliverable_project/models/project.py:40]: I am not sure about this constraint anymore cc @seb-elico

**Review by elicoidal — COMMENTED** [review]
Almost there. Can you prepare the README too and I will review it

**Comment by elicoidal** [comment]
@JoJoJoJoJoJoJo @joycehuan @seb-elico please have a look

File: business_requirement_deliverable_project/models/business.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            return {
+                'type': 'ir.actions.act_window',
+                'view_type': 'form',
+                'view_mode': 'tree,form,graph',
+                'res_model': 'project.project',
+                'target': 'current',
+                'domain': domain
+            }
+
+    all_project_generated = fields.Boolean(
```
Comment (JoJoJoJoJoJoJo) [inline business_requirement_deliverable_project/models/business.py:69]: Move the field up for convenience.

**Review by JoJoJoJoJoJoJo — APPROVED** [review]
LGTM

File: business_requirement_deliverable_project/README.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/README.rst:3]: LGPL-3

File: business_requirement_deliverable_project/README.rst [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+==============================
+Business Requirement - Project
+==============================
+
+Introduction
+============
+
+This module is part of a set of modules (`Business Requirements <https://github.com/OCA/business-requirement/blob/10.0/README.md>`_)
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/README.rst:12]: 10.0 --> 12.0

File: business_requirement_deliverable_project/README.rst [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+* 7 tasks are created for
+
+  * in project linked to DL1 (2 tasks for RL1 and RL2),
+  * in project linked to DL2 (2 tasks for RL3~4),
+  * in project linked to DL3 (2 tasks for RL5~6) and
+  * in project linked to DL4 (1 task for RL7)
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/222/10.0
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/README.rst:175]: same herer

File: business_requirement_deliverable_project/wizard/generate_projects_view.xml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                                   attrs="{'readonly': [('for_br', '!=', True)]}"/>
+                        </group>
+                    </group>
+                </sheet>
+            </form>
+            <footer>
+                <button name="apply" string="_Apply" type="object"
+                        class="oe_highlight"/>
+                or
+                <button string="_Cancel" class="oe_link" special="cancel"/>
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/wizard/generate_projects_view.xml:54]: string="Cancel"

File: business_requirement_deliverable_project/wizard/generate_projects_view.xml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                        <br/>
+                        <group col="4">
+                            <field name="for_childs"
+                                   attrs="{'readonly': [('for_br', '!=', True)]}"/>
+                        </group>
+                    </group>
+                </sheet>
+            </form>
+            <footer>
+                <button name="apply" string="_Apply" type="object"
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/wizard/generate_projects_view.xml:51]: string="Apply"

File: business_requirement_deliverable_project/views/business_view.xml [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+        <field name="view_mode">tree,kanban,form,graph</field>
+        <field name="view_type">form</field>
+        <field name="context">{}</field>
+    </record>
+    <record model="ir.ui.view" id="edit_project">
+        <field name="name">project.project.form</field>
+        <field name="model">project.project</field>
+        <field name="inherit_id" ref="project.edit_project"/>
+        <field name="arch" type="xml">
+            <xpath expr="//form/sheet/div[@name='button_box']"
```
Comment (nikul-serpentcs) [inline business_requirement_deliverable_project/views/business_view.xml:159]: `//div[@name='button_box']` is enough for xpath

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improve code

**Comment by pedrobaeza** [comment]
Closing as old and not green

**Pull request closed without merging**